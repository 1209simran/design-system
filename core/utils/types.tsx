export type BaseProps = {
  className?: string,
  'data-test'?: string,
};

const baseProps = [
  'className',
  'data-test'
];

export const extractBaseProps = (props: Record<string, any>) => {
  const basePropsObj = baseProps.reduce((acc, curr) => {
    return (
      props[curr] ? { ...acc, [curr]: props[curr] } : { ...acc }
    );
  }, {});

  return basePropsObj;
};
