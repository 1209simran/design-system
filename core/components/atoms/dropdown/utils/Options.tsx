export let dropdownOptions: any[] = [];
export const group1: any[] = [];
const group2: any[] = [];

for (let i = 1; i <= 40; i++) {
  group1.push({
    label: `Option ${i}`,
    value: `Option ${i}`,
  });
}

for (let i = 41; i <= 50; i++) {
  group2.push({
    label: `Option ${i}`,
    value: `Option ${i}`,
  });
}

dropdownOptions.push({
  group: true,
  label: 'Group 1',
  items: group1,
});

dropdownOptions.push({
  group: true,
  label: 'Group 2',
  items: group2,
});

export const selectedStoryOptions: any[] = [];

for (let i = 1; i <= 10; i++) {
  selectedStoryOptions.push({
    label: `Option ${i}`,
    value: `Option ${i}`,
    selected: i === 2,
  });
}

export const multiSelectedStoryOptions: any[] = [];

for (let i = 1; i <= 10; i++) {
  multiSelectedStoryOptions.push({
    label: `Option ${i}`,
    value: `Option ${i}`,
    selected: i === 2 || i === 7,
  });
}

export const storyOptions = group1.slice(0, 10);

export const storyWrapOptions = [
  {
    label: 'Design System Dropdown',
    value: 'Design System Dropdown'
  },
  {
    label: 'UI Kit Dropdown',
    value: 'UI Kit Dropdown',
  },
  {
    label: 'Innovaccer Analytics',
    value: 'Innovaccer Analytics'
  }
];

export const subInfoItems = [
  {
    label: 'Option 1',
    value: 'Option 1',
    subInfo: 'subInfo'
  },
  {
    label: 'Option 2',
    value: 'Option 2',
    subInfo: 'subInfo'
  },
  {
    label: 'Option 3',
    value: 'Option 3',
    subInfo: 'subInfo',
  },
];

export const iconItems = [
  {
    label: 'Option 1',
    value: 'Option 1',
    icon: 'events'
  },
  {
    label: 'Option 2',
    value: 'Option 2',
    icon: 'events',
  },
  {
    label: 'Option 3',
    value: 'Option 3',
    icon: 'events',
  },
];
