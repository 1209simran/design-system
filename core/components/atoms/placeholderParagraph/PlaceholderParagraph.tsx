import * as React from 'react';
import classNames from 'classnames';

export type Sizes = 'small' | 'medium' | 'large';

export interface IPlaceholderParagraphProps {
  length?: Sizes;
}

const Paragraph: React.FunctionComponent<IPlaceholderParagraphProps> = props => {
  const { length = 'medium' } = props;

  const classes = classNames({
    'Placeholder-paragraph': true,
    'Placeholder--animation': true,
    [`Placeholder-paragraph--${length}`]: length,
  });

  return (
    <div className={classes} />
  );
};

export default Paragraph;
