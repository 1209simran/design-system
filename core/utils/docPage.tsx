// @ts-nocheck

import * as React from 'react';
import { Title, Props, Description, Primary, Story, PropsProps } from '@storybook/addon-docs/blocks';

interface DocPageProps {
  title?: string;
  description?: React.ReactNode;
  props?: PropsProps;
}

export const docPage = (props: DocPageProps) => {
  const {
    title,
    description,
    props,
    stories
  } = props;

  const separatorIndex = title?.indexOf('|');

  return (
    <>
      <Title>{title && separatorIndex !== -1 ? title.slice(separatorIndex + 1) : title}</Title>
      <Description>
        {description}
      </Description>
      {/* <Primary /> */}
      {/* <Story id="switch-state--alert"  /> */}
      <Props {...props} />
    </>
  );
};

export default docPage;
