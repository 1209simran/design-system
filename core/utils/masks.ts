import { Mask } from '@/utils/types';
import { DatePickerProps } from "@/index.type";

type DateFormat = DatePickerProps['inputFormat'];
export const date: Record<DateFormat, Mask> = {
  'dd/mm/yyyy': [/[0123]/, /\d/, '/', /\[01]/, /\d/, '/', /\d/, /\d/, /\d/, /\d/],
  'mm/dd/yyyy': [/[01]/, /\d/, '/', /[0123]/, /\d/, '/', /\d/, /\d/, /\d/, /\d/],
  'yyyy/mm/dd': [/\d/, /\d/, /\d/, /\d/, '/', /[01]/, /\d/, '/', /[0123]/, /\d/],
  'dd-mm-yyyy': [/[0123]/, /\d/, '-', /[01]/, /\d/, '-', /\d/, /\d/, /\d/, /\d/],
  'mm-dd-yyyy': [/[01]/, /\d/, '-', /[0123]/, /\d/, '-', /\d/, /\d/, /\d/, /\d/],
  'yyyy-mm-dd': [/\d/, /\d/, /\d/, /\d/, '-', /[01]/, /\d/, '-', /[0123]/, /\d/],
};
