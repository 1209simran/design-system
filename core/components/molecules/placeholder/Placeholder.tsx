import * as React from 'react';

import PlaceholderImage, { Size } from '@/components/atoms/placeholderImage';
import { PlaceholderParagraphProps } from '@/components/atoms/placeholderParagraph';

export interface PlaceholderProps {
  /**
   * Shows `Placeholder` along with image
   */
  withImage?: boolean;
  /**
   * Changes shape of `Placeholder` to circle
   */
  round?: boolean;
  /**
   * Specifies dimension of `Placeholder`
   * @default "small"
   */
  imageSize?: Size;
  /**
   * Adds CSS
   */
  style?: React.CSSProperties;
  /**
   * To be rendered in `Placeholder` wrapper
   */
  children?: React.ReactElement<PlaceholderParagraphProps> | React.ReactElement<PlaceholderParagraphProps>[];
}

export const Placeholder = (props: PlaceholderProps) => {
  const {
    imageSize,
    withImage,
    round,
    children,
  } = props;

  return (
    <div className="Placeholder">
      {withImage && (
        <div className="Placeholder-image">
          <PlaceholderImage round={round} size={imageSize} />
        </div>
      )}
      <div className="Placeholder-paragraph">
        {children}
      </div>
    </div>
  );
};

Placeholder.defaultProps = {
  withImage: true,
  imageSize: 'small',
};

Placeholder.displayName = 'Placeholder';

export default Placeholder;
