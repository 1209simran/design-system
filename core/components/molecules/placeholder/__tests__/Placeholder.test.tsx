import * as React from 'react';
import { shallow } from 'enzyme';
import Placeholder, { IPlaceholderProps as IProps } from '../Placeholder';
import PlaceholderParagraph from '@/components/atoms/placeholderParagraph';
import { TestHelper, filterUndefined, valueHelper, testMessageHelper } from '@/utils/TestHelper';

const Style = [{
  width: valueHelper('150px', { required: true }),
  height: valueHelper('150px', { required: true })
}];
const Size = ['small', 'medium', 'large'];
const BooleanValue = [true, false];

const Mapper = {
  withImage: valueHelper(BooleanValue, { iterate: true }),
  round: valueHelper(BooleanValue, { iterate: true }),
  imageSize: valueHelper(Size, { iterate: true }),
  style: valueHelper(Style, { iterate: true }),
};

describe('Placeholder component', () => {
  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as IProps;

    it(testMessageHelper(attr), () => {
      const tree = shallow(
        <Placeholder {...attr}>
          <PlaceholderParagraph length="small" />
        </Placeholder>
      );
      expect(tree).toMatchSnapshot();
    });
  };

  TestHelper(Mapper, testFunc);
});
