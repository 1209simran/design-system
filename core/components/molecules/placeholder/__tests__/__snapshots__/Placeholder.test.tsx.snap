// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Placeholder component 
  
 1`] = `
<div
  className="Placeholder"
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  imageSize: "large"
 1`] = `
<div
  className="Placeholder"
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  imageSize: "large", style: {"width":"150px","height":"150px"}
 1`] = `
<div
  className="Placeholder"
  style={
    Object {
      "height": "150px",
      "width": "150px",
    }
  }
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  imageSize: "medium"
 1`] = `
<div
  className="Placeholder"
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  imageSize: "medium", style: {"width":"150px","height":"150px"}
 1`] = `
<div
  className="Placeholder"
  style={
    Object {
      "height": "150px",
      "width": "150px",
    }
  }
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  imageSize: "small"
 1`] = `
<div
  className="Placeholder"
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  imageSize: "small", style: {"width":"150px","height":"150px"}
 1`] = `
<div
  className="Placeholder"
  style={
    Object {
      "height": "150px",
      "width": "150px",
    }
  }
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  round: false
 1`] = `
<div
  className="Placeholder"
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  round: false, imageSize: "large"
 1`] = `
<div
  className="Placeholder"
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  round: false, imageSize: "large", style: {"width":"150px","height":"150px"}
 1`] = `
<div
  className="Placeholder"
  style={
    Object {
      "height": "150px",
      "width": "150px",
    }
  }
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  round: false, imageSize: "medium"
 1`] = `
<div
  className="Placeholder"
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  round: false, imageSize: "medium", style: {"width":"150px","height":"150px"}
 1`] = `
<div
  className="Placeholder"
  style={
    Object {
      "height": "150px",
      "width": "150px",
    }
  }
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  round: false, imageSize: "small"
 1`] = `
<div
  className="Placeholder"
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  round: false, imageSize: "small", style: {"width":"150px","height":"150px"}
 1`] = `
<div
  className="Placeholder"
  style={
    Object {
      "height": "150px",
      "width": "150px",
    }
  }
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  round: false, style: {"width":"150px","height":"150px"}
 1`] = `
<div
  className="Placeholder"
  style={
    Object {
      "height": "150px",
      "width": "150px",
    }
  }
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  round: true
 1`] = `
<div
  className="Placeholder"
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  round: true, imageSize: "large"
 1`] = `
<div
  className="Placeholder"
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  round: true, imageSize: "large", style: {"width":"150px","height":"150px"}
 1`] = `
<div
  className="Placeholder"
  style={
    Object {
      "height": "150px",
      "width": "150px",
    }
  }
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  round: true, imageSize: "medium"
 1`] = `
<div
  className="Placeholder"
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  round: true, imageSize: "medium", style: {"width":"150px","height":"150px"}
 1`] = `
<div
  className="Placeholder"
  style={
    Object {
      "height": "150px",
      "width": "150px",
    }
  }
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  round: true, imageSize: "small"
 1`] = `
<div
  className="Placeholder"
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  round: true, imageSize: "small", style: {"width":"150px","height":"150px"}
 1`] = `
<div
  className="Placeholder"
  style={
    Object {
      "height": "150px",
      "width": "150px",
    }
  }
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  round: true, style: {"width":"150px","height":"150px"}
 1`] = `
<div
  className="Placeholder"
  style={
    Object {
      "height": "150px",
      "width": "150px",
    }
  }
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  style: {"width":"150px","height":"150px"}
 1`] = `
<div
  className="Placeholder"
  style={
    Object {
      "height": "150px",
      "width": "150px",
    }
  }
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: false
 1`] = `
<div
  className="Placeholder"
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: false, imageSize: "large"
 1`] = `
<div
  className="Placeholder"
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: false, imageSize: "large", style: {"width":"150px","height":"150px"}
 1`] = `
<div
  className="Placeholder"
  style={
    Object {
      "height": "150px",
      "width": "150px",
    }
  }
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: false, imageSize: "medium"
 1`] = `
<div
  className="Placeholder"
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: false, imageSize: "medium", style: {"width":"150px","height":"150px"}
 1`] = `
<div
  className="Placeholder"
  style={
    Object {
      "height": "150px",
      "width": "150px",
    }
  }
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: false, imageSize: "small"
 1`] = `
<div
  className="Placeholder"
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: false, imageSize: "small", style: {"width":"150px","height":"150px"}
 1`] = `
<div
  className="Placeholder"
  style={
    Object {
      "height": "150px",
      "width": "150px",
    }
  }
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: false, round: false
 1`] = `
<div
  className="Placeholder"
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: false, round: false, imageSize: "large"
 1`] = `
<div
  className="Placeholder"
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: false, round: false, imageSize: "large", style: {"width":"150px","height":"150px"}
 1`] = `
<div
  className="Placeholder"
  style={
    Object {
      "height": "150px",
      "width": "150px",
    }
  }
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: false, round: false, imageSize: "medium"
 1`] = `
<div
  className="Placeholder"
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: false, round: false, imageSize: "medium", style: {"width":"150px","height":"150px"}
 1`] = `
<div
  className="Placeholder"
  style={
    Object {
      "height": "150px",
      "width": "150px",
    }
  }
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: false, round: false, imageSize: "small"
 1`] = `
<div
  className="Placeholder"
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: false, round: false, imageSize: "small", style: {"width":"150px","height":"150px"}
 1`] = `
<div
  className="Placeholder"
  style={
    Object {
      "height": "150px",
      "width": "150px",
    }
  }
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: false, round: false, style: {"width":"150px","height":"150px"}
 1`] = `
<div
  className="Placeholder"
  style={
    Object {
      "height": "150px",
      "width": "150px",
    }
  }
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: false, round: true
 1`] = `
<div
  className="Placeholder"
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: false, round: true, imageSize: "large"
 1`] = `
<div
  className="Placeholder"
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: false, round: true, imageSize: "large", style: {"width":"150px","height":"150px"}
 1`] = `
<div
  className="Placeholder"
  style={
    Object {
      "height": "150px",
      "width": "150px",
    }
  }
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: false, round: true, imageSize: "medium"
 1`] = `
<div
  className="Placeholder"
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: false, round: true, imageSize: "medium", style: {"width":"150px","height":"150px"}
 1`] = `
<div
  className="Placeholder"
  style={
    Object {
      "height": "150px",
      "width": "150px",
    }
  }
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: false, round: true, imageSize: "small"
 1`] = `
<div
  className="Placeholder"
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: false, round: true, imageSize: "small", style: {"width":"150px","height":"150px"}
 1`] = `
<div
  className="Placeholder"
  style={
    Object {
      "height": "150px",
      "width": "150px",
    }
  }
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: false, round: true, style: {"width":"150px","height":"150px"}
 1`] = `
<div
  className="Placeholder"
  style={
    Object {
      "height": "150px",
      "width": "150px",
    }
  }
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: false, style: {"width":"150px","height":"150px"}
 1`] = `
<div
  className="Placeholder"
  style={
    Object {
      "height": "150px",
      "width": "150px",
    }
  }
>
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: true
 1`] = `
<div
  className="Placeholder Placeholder--withImage"
>
  <Image
    imageSize="small"
  />
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: true, imageSize: "large"
 1`] = `
<div
  className="Placeholder Placeholder--withImage"
>
  <Image
    imageSize="large"
  />
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: true, imageSize: "large", style: {"width":"150px","height":"150px"}
 1`] = `
<div
  className="Placeholder Placeholder--withImage"
  style={
    Object {
      "height": "150px",
      "width": "150px",
    }
  }
>
  <Image
    imageSize="large"
  />
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: true, imageSize: "medium"
 1`] = `
<div
  className="Placeholder Placeholder--withImage"
>
  <Image
    imageSize="medium"
  />
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: true, imageSize: "medium", style: {"width":"150px","height":"150px"}
 1`] = `
<div
  className="Placeholder Placeholder--withImage"
  style={
    Object {
      "height": "150px",
      "width": "150px",
    }
  }
>
  <Image
    imageSize="medium"
  />
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: true, imageSize: "small"
 1`] = `
<div
  className="Placeholder Placeholder--withImage"
>
  <Image
    imageSize="small"
  />
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: true, imageSize: "small", style: {"width":"150px","height":"150px"}
 1`] = `
<div
  className="Placeholder Placeholder--withImage"
  style={
    Object {
      "height": "150px",
      "width": "150px",
    }
  }
>
  <Image
    imageSize="small"
  />
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: true, round: false
 1`] = `
<div
  className="Placeholder Placeholder--withImage"
>
  <Image
    imageSize="small"
    round={false}
  />
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: true, round: false, imageSize: "large"
 1`] = `
<div
  className="Placeholder Placeholder--withImage"
>
  <Image
    imageSize="large"
    round={false}
  />
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: true, round: false, imageSize: "large", style: {"width":"150px","height":"150px"}
 1`] = `
<div
  className="Placeholder Placeholder--withImage"
  style={
    Object {
      "height": "150px",
      "width": "150px",
    }
  }
>
  <Image
    imageSize="large"
    round={false}
  />
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: true, round: false, imageSize: "medium"
 1`] = `
<div
  className="Placeholder Placeholder--withImage"
>
  <Image
    imageSize="medium"
    round={false}
  />
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: true, round: false, imageSize: "medium", style: {"width":"150px","height":"150px"}
 1`] = `
<div
  className="Placeholder Placeholder--withImage"
  style={
    Object {
      "height": "150px",
      "width": "150px",
    }
  }
>
  <Image
    imageSize="medium"
    round={false}
  />
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: true, round: false, imageSize: "small"
 1`] = `
<div
  className="Placeholder Placeholder--withImage"
>
  <Image
    imageSize="small"
    round={false}
  />
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: true, round: false, imageSize: "small", style: {"width":"150px","height":"150px"}
 1`] = `
<div
  className="Placeholder Placeholder--withImage"
  style={
    Object {
      "height": "150px",
      "width": "150px",
    }
  }
>
  <Image
    imageSize="small"
    round={false}
  />
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: true, round: false, style: {"width":"150px","height":"150px"}
 1`] = `
<div
  className="Placeholder Placeholder--withImage"
  style={
    Object {
      "height": "150px",
      "width": "150px",
    }
  }
>
  <Image
    imageSize="small"
    round={false}
  />
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: true, round: true
 1`] = `
<div
  className="Placeholder Placeholder--withImage"
>
  <Image
    imageSize="small"
    round={true}
  />
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: true, round: true, imageSize: "large"
 1`] = `
<div
  className="Placeholder Placeholder--withImage"
>
  <Image
    imageSize="large"
    round={true}
  />
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: true, round: true, imageSize: "large", style: {"width":"150px","height":"150px"}
 1`] = `
<div
  className="Placeholder Placeholder--withImage"
  style={
    Object {
      "height": "150px",
      "width": "150px",
    }
  }
>
  <Image
    imageSize="large"
    round={true}
  />
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: true, round: true, imageSize: "medium"
 1`] = `
<div
  className="Placeholder Placeholder--withImage"
>
  <Image
    imageSize="medium"
    round={true}
  />
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: true, round: true, imageSize: "medium", style: {"width":"150px","height":"150px"}
 1`] = `
<div
  className="Placeholder Placeholder--withImage"
  style={
    Object {
      "height": "150px",
      "width": "150px",
    }
  }
>
  <Image
    imageSize="medium"
    round={true}
  />
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: true, round: true, imageSize: "small"
 1`] = `
<div
  className="Placeholder Placeholder--withImage"
>
  <Image
    imageSize="small"
    round={true}
  />
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: true, round: true, imageSize: "small", style: {"width":"150px","height":"150px"}
 1`] = `
<div
  className="Placeholder Placeholder--withImage"
  style={
    Object {
      "height": "150px",
      "width": "150px",
    }
  }
>
  <Image
    imageSize="small"
    round={true}
  />
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: true, round: true, style: {"width":"150px","height":"150px"}
 1`] = `
<div
  className="Placeholder Placeholder--withImage"
  style={
    Object {
      "height": "150px",
      "width": "150px",
    }
  }
>
  <Image
    imageSize="small"
    round={true}
  />
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;

exports[`Placeholder component 
  withImage: true, style: {"width":"150px","height":"150px"}
 1`] = `
<div
  className="Placeholder Placeholder--withImage"
  style={
    Object {
      "height": "150px",
      "width": "150px",
    }
  }
>
  <Image
    imageSize="small"
  />
  <div
    className="ml-4 w-100"
  >
    <Paragraph
      length="small"
    />
  </div>
</div>
`;
